rfcs/header-navigation-component.md

```diff
@@ -0,0 +1,56 @@
+# Header Navigation Component
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {HeaderNavigation} from 'baseui/header-navigation';
+import {Button, KIND} from 'baseui/button';
+
+export default () => <HeaderNavigation homeElement={<span>Uber</span>}>
+  <a href="/a">Link One</a>
+  <a href="/b">Link Two</a>
+  <Button onClick={() => console.log('clicked')} kind={KIND.primary}>Get started</Button>
+</HeaderNavigation>;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {HeaderNavigation} from 'baseui/header-navigation';
+import {Button, KIND} from 'baseui/button';
+import {StatefulMenu as Menu} from 'baseui/menu';
+const ITEMS = [{label: 'menu item 1'}, {label: 'menu item 2'}];
+
+export default () => <HeaderNavigation homeElement={<Menu items={ITEMS} />}
```

Comment:
Using `Menu` here directly seems like it would just render a broken or janky header (unless I'm missing something), maybe we should use another example so users aren't confused?

Reply:
Yes, it won't work like this. Need another example